src/Avalonia.Controls/Button.cs

```diff
@@ -251,7 +251,9 @@ protected override void OnPointerReleased(PointerReleasedEventArgs e)
                 IsPressed = false;
                 e.Handled = true;
 
-                if (ClickMode == ClickMode.Release && new Rect(Bounds.Size).Contains(e.GetPosition(this)))
+                //only renderer (hittesting) know better whether pointer is over the bounds of the button
+                if (ClickMode == ClickMode.Release && 
+                    (IsPointerOver || new Rect(Bounds.Size).Contains(e.GetPosition(this))))
```

Comment:
Is the check on `Bounds` still necessary here? Wouldn't just checking `IsPointerOver` be sufficient?

Reply:
@grokys 
On theory it shouldn't be nesessary, but in real life it might be needed for mobile platforms as i'm not sure pointer over is raised for touch or not. It's bit more safe to keep Bounds.Contains(positions) probably.
 I see some time ago it was changed to work like this (original implementation was working as only pointerover check) may be for a reason. check this commit: https://github.com/AvaloniaUI/Avalonia/commit/4d30de7c09b3b39efaf7a2c622a84bb4df56af89#diff-45a4dd48c9a2f83d7def2fd422d1423c

I'm fine if we check only IsPointerOver, but you guys might know better, i'm good either way.

Reply:
Ah yes, I remember this now. From the PR that introduced that change:

> Previously a button click was carried out when :pointerover was set, but because the button captures mouse input on mouse down, this is always true while the mouse is held. Check that the pointer was released within the bounds of the control.

`:pointerover` and `IsPointerOver` are the same thing, so making the change in this PR would result in a return of #938.

I think we'll need to do a hit-test here.

Reply:
hm, yes sound about right i've missed that pointer is always over when mouse is captured

Reply:
@grokys hit testing in button is resolved, altough for a complete click+rendertransform+ui fix #2002  is needed, as int this pr button (with rendertransform) mouse over/click look funky

Reply:
This sound about right, haven't noticed that mouse is over always while button is captured.